lib/services/local/usagereporter.go

```diff
@@ -0,0 +1,544 @@
+			},
+		}, nil
+	case *services.UsageUIOnboardAddFirstResourceClickEvent:
+		return &prehogapi.SubmitEventRequest{
+			ClusterName: clusterName,
+			Timestamp:   time,
+			Event: &prehogapi.SubmitEventRequest_UiOnboardAddFirstResourceClick{
+				UiOnboardAddFirstResourceClick: (*prehogapi.UIOnboardAddFirstResourceClickEvent)(e),
+			},
+		}, nil
+	case *services.UsageUIOnboardAddFirstResourceLaterClickEvent:
+		return &prehogapi.SubmitEventRequest{
+			ClusterName: clusterName,
+			Timestamp:   time,
+			Event: &prehogapi.SubmitEventRequest_UiOnboardAddFirstResourceLaterClick{
+				UiOnboardAddFirstResourceLaterClick: (*prehogapi.UIOnboardAddFirstResourceLaterClickEvent)(e),
+			},
+		}, nil
+	case *services.UsageUIOnboardSetCredentialSubmit:
+		return &prehogapi.SubmitEventRequest{
+			ClusterName: clusterName,
+			Timestamp:   time,
+			Event: &prehogapi.SubmitEventRequest_UiOnboardSetCredentialSubmit{
+				UiOnboardSetCredentialSubmit: (*prehogapi.UIOnboardSetCredentialSubmitEvent)(e),
+			},
+		}, nil
+	case *services.UsageUIOnboardRegisterChallengeSubmit:
+		return &prehogapi.SubmitEventRequest{
+			ClusterName: clusterName,
+			Timestamp:   time,
+			Event: &prehogapi.SubmitEventRequest_UiOnboardRegisterChallengeSubmit{
+				UiOnboardRegisterChallengeSubmit: (*prehogapi.UIOnboardRegisterChallengeSubmitEvent)(e),
+			},
+		}, nil
+	case *services.UsageUIOnboardRecoveryCodesContinueClick:
+		return &prehogapi.SubmitEventRequest{
+			ClusterName: clusterName,
+			Timestamp:   time,
+			Event: &prehogapi.SubmitEventRequest_UiOnboardRecoveryCodesContinueClick{
+				UiOnboardRecoveryCodesContinueClick: (*prehogapi.UIOnboardRecoveryCodesContinueClickEvent)(e),
+			},
+		}, nil
+	default:
+		return nil, trace.BadParameter("unexpected event usage type %T", event)
+	}
+}
+
+func (r *UsageReporter) SubmitAnonymizedUsageEvents(events ...services.UsageAnonymizable) error {
+	var anonymized []*prehogapi.SubmitEventRequest
+
+	for _, e := range events {
+		e.Anonymize(r.anonymizer)
+
+		converted, err := r.convertEvent(e)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		anonymized = append(anonymized, converted)
+
+		usageEventsSubmitted.Inc()
+	}
+
+	r.events <- anonymized
+
+	return nil
+}
+
+// resubmitEvents resubmits events that have already been processed (in case of
+// some error during submission).
+func (r *UsageReporter) resubmitEvents(events []*prehogapi.SubmitEventRequest) {
+	usageEventsRequeuedTotal.Add(float64(len(events)))
+
+	r.events <- events
+}
+
+func NewPrehogSubmitter(ctx context.Context, prehogEndpoint string, clientCert *tls.Certificate, caCertPEM []byte) (UsageSubmitFunc, error) {
+	tlsConfig := &tls.Config{
+		// Self-signed test licenses may not have a proper issuer and won't be
+		// used if just passed in via Certificates, so we'll use this to
+		// explicitly set the client cert we want to use.
+		GetClientCertificate: func(cri *tls.CertificateRequestInfo) (*tls.Certificate, error) {
```

Comment:
If we're always going to return the same client cert, why use the `GetClientCertificate` callback? This is useful if you need information from the TLS handshake in order to select the appropriate cert, which doesn't seem to be the case here.

Can't we just set `(tls.Config).Certificates` instead?

Reply:
When testing locally we found that Go's HTTP client will opt not to send the client cert in all cases due to an issuer mismatch when using self signed licenses, i.e. the client would opt not to send a cert prehog would otherwise trust. This method seems to work okay both in prod (with sane cert issuers) and locally.

Reply:
Gotcha. If that's the case I would just put that in a comment so someone like me doesn't come around and break it by trying to simplify things :-)

Reply:
There is a small note just above :) 
```go
		// Self-signed test licenses may not have a proper issuer and won't be
		// used if just passed in via Certificates, so we'll use this to
		// explicitly set the client cert we want to use.
```